internal/wsproxy/conn_test.go

```diff
@@ -82,47 +83,249 @@ func (m *mockRecorder) IsHeaderWritten() bool {
 	return m.headerWritten
 }
 
+type WebSocketMessage struct {
+	Bytes   []byte // raw bytes of the websocket frame
+	Payload []byte // the payload of the websocket frame
+}
```

Comment:
Perhaps we could avoid duplicating the payload with something like
```go
type WebSocketMessage struct {
	Headers   []byte // raw bytes of the websocket frame
	Payload []byte // the payload of the websocket frame
}

func (m WebSocketMessage)Raw() {
	return append(m.Headers, m.Payload)
}
```
?

Reply:
if the message contains a masked payload, then the raw bytes cannot be headers + payload appended, the raw bytes should have the masked payload and the payload we are asserting should be the unmasked final data